dessia_common/datatools/dataset.py

```diff
@@ -669,11 +669,18 @@ def _histogram_unic_value(self, idx_col: int, name_attr: str):
     def _tooltip_attributes(self):
         return self.common_attributes
 
-    def _plot_data_list(self):
-        plot_data_list = []
-        for row, _ in enumerate(self.dessia_objects):
-            plot_data_list.append({attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)})
-        return plot_data_list
+    def _object_to_sample(self, dessia_object: DessiaObject, row: int):
```

Comment:
I would go even further that that. Consider this : 

```python
def _object_to_sample(self, dessia_object: DessiaObject, row: int, label: str = None) -> Sample:
    sample_values = {attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)}
    if label is not None:
        sample_values["Cluster Label"] = label
        reference_path = f"#/dessia_objects/{row}"
        name = dessia_object.name if dessia_object.name else f"Sample {row}"
    return Sample(values=sample_values, reference_path=reference_path, name=name)

# DATASET
def _to_samples(self):
    return [self._object_to_sample(dessia_object=object_, row=row)
            for row, object_ in enumerate(self.dessia_objects)]
    
# CLUSTER
def _to_samples(self):
     return [self._object_to_sample(dessia_object=object_, row=row, label=label)
             for row, (object_, label) in enumerate(zip(self.dessia_objects, self.labels))]
```

Notes : 
- Normally, I would have called loop variables inside comprehension list as one letters, withtout naming function arguments like this : 
`samples = [object_to_samples(o, i, l) for i, (o, l) in enumerate(zip(self.objects, self.labels))]`
BUT, I am annoyed here, because "l" is forbidden as a one-letter variable name, as it is easily confused with a 1 or a I depending on the font, so that is why I named them without abbreviating AND I named the arguments for clarity (it is "consistence of the length" in a way, you either go full short or full complete)

- When you have more than one arg in a function call or an object instantiation, prefer naming args in the statement. For ex :
` Sample(values=sample_values, reference_path=reference_path, name=name)`. this is because it is easier to read and understand and is robust to any refactor. If we add, remove or change orders of arguments, the code will still work as long as there weren't any breaking changes. 
BUT not when there is only one, because it is normally straightforward AND it can cause a bug when the code is compiled with Cython. ex : 
`object_._display_from_selector(selector)`
and it is not that subject to refactor as there is only one arg

Reply:
Note, that if you are inheriting your clusters with Dataset, you could, in my code, instead of giving a label arg to the `_object_to_sample` method that is never used in the case of the Dataset, make use of the inheritance. What I mean is do the following : 

- Define your base `_object_to_sample` and `_to_samples` method in Dataset
- Make Cluster inherit from Dataset
- Overwrite `_object_to_sample` method
- Inside, call the super() `_object_to_sample` method,
- Loop on a zip of it and self.labels, in order to change already created values
- Never overwrite `_to_samples` method

That might be the right way to do it

Reply:
Like this : 
```python

class Dataset:
    def __init__(self):
        self.common_attributes = []
        self.matrix = [[]]
        self.dessia_objects = []
        self.labels = []
        pass

    def _object_to_sample(self, dessia_object: DessiaObject, row: int) -> Sample:
        sample_values = {attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)}
        reference_path = f"#/dessia_objects/{row}"
        name = dessia_object.name if dessia_object.name else f"Sample {row}"
        return Sample(values=sample_values, reference_path=reference_path, name=name)

    # DATASET
    def _to_samples(self):
        return [self._object_to_sample(dessia_object=object_, row=row)
                for row, object_ in enumerate(self.dessia_objects)]


class Cluster(Dataset):
    def __init__(self):
        Dataset.__init__(self)

    def _object_to_sample(self, dessia_object: DessiaObject, row: int) -> Sample:
        samples = super(Cluster, self)._object_to_sample(dessia_object=dessia_object, row=row)
        for sample, label in zip(samples, self.labels):
            sample.values["Cluster Label"] = label
        return samples
```

Reply:
I propose to make approximately what is proposed in the first idea. Without rewriting _to_sample method.